errorlint/lint.go

```diff
@@ -189,6 +189,27 @@ func LintErrorComparisons(fset *token.FileSet, info *TypesInfoExt) []Lint {
 			continue
 		}
 
+		hasErrorCase := false
+		for _, caseBlock := range switchStmt.Body.List {
+			caseClause, ok := caseBlock.(*ast.CaseClause)
+			if !ok {
+				continue
+			}
+			for _, clause := range caseClause.List {
+				switch clause := clause.(type) {
+				case nil:
+					continue // default label is safe
+				case *ast.Ident:
+					if clause.Name != "nil" {
+						hasErrorCase = true
+					}
+				}
+			}
+		}
+		if !hasErrorCase {
+			continue
```

Comment:
If I understand this new logic correctly, switch statements will be completely ignored when a nil clause is present?

That does not seem correct to me. I think it would be best to only ignore if a nil is present as the only clause, with or without default.

Reply:
I think you misunderstood. If you did not, then I wrote it wrong. :) The intent is to report the switch only when it actually compares an error by value. That is when it has at least one CaseClause.List entry that represents neither `case nil:` or `default:`.

Reply:
Are the things it should not report already present in the test suite? If not, could you add them?

Reply:
Already done:
- [dc92118](https://github.com/polyfloyd/go-errorlint/pull/27/commits/dc9211803bd701d785798f87a4664aed59957f62) added CompareSwitchNilDefault that emits no reports despite being a `switch err { ... }`. 
- [d7bea88](https://github.com/polyfloyd/go-errorlint/pull/27/commits/d7bea88ac255a268211e86c8142fe1be4aa92fab) added to CompareSwitch to validate the presence of `case nil:` didn't prevent its existing report.